### DISCUSSION TITLE
Is it alllowed to use the same names for groups and hosts

### DISCUSSION THREAD

**Discussion by bessarabov** [umputun/spot#47] [General]
Do you think that this should be allowed, or such inventory should fail spot?

(here `g2` is the name of the host and the name of the group:

```yaml
bessarabov@bessarabov-osx:~$ cat /Users/bessarabov/.spot_inventory.yaml
groups:
  g1:
    - { host: "192.168.31.101", name: "h1" }
    - { host: "192.168.31.102", name: "g2" }

  g2:
    - { host: "192.168.31.86", user: "pi", name: "h3" }
bessarabov@bessarabov-osx:~$ spot --inventory=/Users/bessarabov/.spot_inventory.yaml --target=g2 date
spot v0.9.1-0aa555f-2023-05-02T08:06:47Z
[h3 192.168.31.86:22] run task "ad-hoc", commands: 1
[h3 192.168.31.86:22]  > sh -c "date"
[h3 192.168.31.86:22]  > Tue 02 May 2023 07:15:35 PM BST
[h3 192.168.31.86:22] completed command "ad-hoc" {script: sh -c "date"} (261ms)
[h3 192.168.31.86:22] completed task "ad-hoc", commands: 1 (386ms)
bessarabov@bessarabov-osx:~$
```

**Comment by umputun** [comment]
This is ok. If the same combo of host+port+user appears in multiple places, it will be deduplicated from the list of destinations automatically.  I added this dedup last night.

Having the same host in multiple groups is probably fine as long as the user understands what he is doing.

**Comment by bessarabov** [comment]
I'm sorry, I don't think that I understand your answer. All the 3 hosts in my example inventory has diffrent ip address.

**Reply by umputun** [reply]
ah, my bad, I misread your question. I thought you were talking about host addresses, not names.

I think such conflicts are user errors in most cases. In some cases, i can see this conflict is intentional, for example, the user has several data centers with the same naming and split it by "datacenter group"

I think we should keep it for now, and will see if this really a problem

**Comment by bessarabov** [comment]
> I think we should keep it for now, and will see if this really a problem

Got it. Thank you, I'm closing this conversation.
